examples/reflection/generic_reflection.rs

```diff
@@ -13,7 +13,7 @@ fn main() {
 }
 
 #[derive(Reflect)]
-struct MyType<T: Reflect> {
+struct MyType<T: Reflect + TypeName> {
```

Comment:
Nit: Since this is an example, could you add a quick doc comment stating why we need to define `TypeName`. I suppose it's not the worst thing if we don't but it could be nice :)